packages/thirdweb/src/extensions/modular/__generated__/ClaimableERC1155/read/encodeBytesBeforeMintERC1155.ts

```diff
@@ -0,0 +1,234 @@
+    components: [
+      {
+        name: "request",
+        type: "tuple",
+        internalType: "struct ClaimableERC1155.ClaimRequestERC1155",
+        components: [
+          {
+            name: "tokenId",
+            type: "uint256",
+            internalType: "uint256",
+          },
+          {
+            name: "startTimestamp",
+            type: "uint48",
+            internalType: "uint48",
+          },
+          {
+            name: "endTimestamp",
+            type: "uint48",
+            internalType: "uint48",
+          },
+          {
+            name: "recipient",
+            type: "address",
+            internalType: "address",
+          },
+          {
+            name: "quantity",
+            type: "uint256",
+            internalType: "uint256",
+          },
+          {
+            name: "currency",
+            type: "address",
+            internalType: "address",
+          },
+          {
+            name: "pricePerUnit",
+            type: "uint256",
+            internalType: "uint256",
+          },
+          {
+            name: "uid",
+            type: "bytes32",
+            internalType: "bytes32",
+          },
+        ],
+      },
+      {
+        name: "signature",
+        type: "bytes",
+        internalType: "bytes",
+      },
+      {
+        name: "currency",
+        type: "address",
+        internalType: "address",
+      },
+      {
+        name: "pricePerUnit",
+        type: "uint256",
+        internalType: "uint256",
+      },
+      {
+        name: "recipientAllowlistProof",
+        type: "bytes32[]",
+        internalType: "bytes32[]",
+      },
+    ],
+  },
+] as const;
+const FN_OUTPUTS = [
+  {
+    name: "",
+    type: "bytes",
+    internalType: "bytes",
+  },
+] as const;
+
+/**
+ * Checks if the `encodeBytesBeforeMintERC1155` method is supported by the given contract.
+ * @param contract The ThirdwebContract.
+ * @returns A promise that resolves to a boolean indicating if the `encodeBytesBeforeMintERC1155` method is supported.
+ * @extension MODULAR
+ * @example
+ * ```ts
+ * import { isEncodeBytesBeforeMintERC1155Supported } from "thirdweb/extensions/modular";
+ *
+ * const supported = await isEncodeBytesBeforeMintERC1155Supported(contract);
+ * ```
+ */
+export async function isEncodeBytesBeforeMintERC1155Supported(
+  contract: ThirdwebContract<any>,
+) {
+  return detectMethod({
+    contract,
+    method: [FN_SELECTOR, FN_INPUTS, FN_OUTPUTS] as const,
+  });
+}
+
+/**
+ * Encodes the parameters for the "encodeBytesBeforeMintERC1155" function.
+ * @param options - The options for the encodeBytesBeforeMintERC1155 function.
+ * @returns The encoded ABI parameters.
+ * @extension MODULAR
+ * @example
+ * ```ts
+ * import { encodeEncodeBytesBeforeMintERC1155Params } "thirdweb/extensions/modular";
+ * const result = encodeEncodeBytesBeforeMintERC1155Params({
+ *  params: ...,
+ * });
+ * ```
+ */
+export function encodeEncodeBytesBeforeMintERC1155Params(
```

Comment:
ideally we add a special case in the generate script to not repeat encode twice

Reply:
the function name on the contract is `encodeBytes...`

we added this prefix for all functions having a bytes param. here's the contract PR: https://github.com/thirdweb-dev/modular-contracts/pull/120/files

cc @jnsdls